src/spikeinterface/extractors/neoextractors/intan.py

```diff
@@ -104,3 +106,106 @@ def _add_channel_groups(self):
+    stream_name : str, default: None
+        If there are several streams, specify the stream name you want to load.
+    all_annotations : bool, default: False
+        Load exhaustively all annotations from neo.
+    use_names_as_ids : bool, default: False
+        Determines the format of the channel IDs used by the extractor. If set to True, the channel IDs will be the
+        names from NeoRawIO. If set to False, the channel IDs will be the ids provided by NeoRawIO.
+    ignore_integrity_checks : bool, default: False
+        If True, data that violates integrity assumptions will be loaded. At the moment the only integrity
+        check we perform is that timestamps are continuous. Setting this to True will ignore this check and set
+        the attribute `discontinuous_timestamps` to True in the underlying neo object.
+
+    Examples
+    --------
+    >>> from spikeinterface.extractors import IntanSplitFilesRecordingExtractor
+    >>> recording = IntanSplitFilesRecordingExtractor("/path/to/intan/folder")
+    """
+
+    def __init__(
+        self,
+        folder_path,
+        mode: Literal["append", "concatenate"] = "concatenate",
+        stream_id=None,
+        stream_name=None,
+        all_annotations=False,
+        use_names_as_ids=False,
+        ignore_integrity_checks: bool = False,
+    ):
+
+        if mode not in ("append", "concatenate"):
+            mode_error = (
+                "Possible options for the `mode` argument are 'concatenate' or 'append', you have entered " f"{mode}."
+            )
+            raise ValueError(mode_error)
+
+        folder_path = Path(folder_path)
+
+        if not folder_path.exists() or not folder_path.is_dir():
+            raise ValueError(f"Folder path {folder_path} does not exist or is not a directory")
+
+        # Find all Intan files
+        file_path_list = [p for p in folder_path.iterdir() if p.suffix.lower() in [".rhd", ".rhs"]]
+
+        if not file_path_list:
+            raise ValueError(f"No Intan files (.rhd or .rhs) found in {folder_path}")
+
+        # Sort files by filename (natural sort)
+        file_path_list.sort(key=lambda x: x.name)
+
+        # Read each file and create recording list
+        recording_list = []
+        for file_path in file_path_list:
+            recording = read_intan(
+                file_path,
+                stream_id=stream_id,
+                stream_name=stream_name,
+                all_annotations=all_annotations,
+                use_names_as_ids=use_names_as_ids,
+                ignore_integrity_checks=ignore_integrity_checks,
+            )
+            recording_list.append(recording)
+
+        # Initialize the parent class with the recording list
+        if mode == "concatenate":
+            ConcatenateSegmentRecording.__init__(self, recording_list)
+        elif mode == "append":
+            AppendSegmentRecording.__init__(self, recording_list)
+
```

Comment:
Is this legal when double hineritance not initialize the other class ?
Just wondering..

Reply:
Yeah, I am weary of that. But maybe it is an opportunity for us to learn something if it blows up at some point x D